dbms/src/Flash/Coprocessor/StreamingDAGResponseWriter.cpp

```diff
@@ -78,8 +78,15 @@ void StreamingDAGResponseWriter<StreamWriterPtr>::finishWrite()
 }
 
 template <class StreamWriterPtr>
-void StreamingDAGResponseWriter<StreamWriterPtr>::write(const Block & block)
+void StreamingDAGResponseWriter<StreamWriterPtr>::write(const Block & block, bool last_null_block)
 {
+    if (last_null_block)
+    {
+        if (rows_in_blocks > 0)
+            encodeThenWriteBlocks<false>();
+        return;
+    }
+
```

Comment:
```suggestion
template <bool last_null_block>
void StreamingDAGResponseWriter<StreamWriterPtr>::write(const Block & block)
{
    if constexpr (last_null_block)
    {
        if (rows_in_blocks > 0)
            encodeThenWriteBlocks<false>();
        return;
    }

```

Reply:
Nice suggestion, extracted a new interface 'flush' instead of reusing the write interface to make it more clear.